### ISSUE TITLE
Connection issues on macOS

### ISSUE THREAD

**Issue by steakscience** [coseyfannitutti/discipline#67]
I'm encountering some connection issues when connecting a Discipline to a Mac.

At the login screen, the keyboard will work. After logging in, the keyboard will stop working and needs to be reconnected. I can reliably reproduce this 100% of the time.

This issue does not happen with other keyboards. It also does not happen with this keyboard on other OSes.

I was wondering if anyone else has had these issues or knows how to fix them. Thanks!

**Comment by BenRoe** [comment]
Never had a similar problem.
Do you use a special firmware, or bootloader?
Try different usb port (without hub) and cables.

**Comment by steakscience** [comment]
> Never had a similar problem.
> Do you use a special firmware, or bootloader?
> Try different usb port (without hub) and cables.

I have tried both: 
- USB C <-> USB C 
- USB A <-> USB C hub

I am (probably) using the latest VIA version, flashed a few months ago

**Comment by BenRoe** [comment]
Try vial. It‘s open source, actively developed and has more features.
https://get.vial.today/

Still a strange behavior. Don’t know what it could be.

**Comment by dhanakane** [comment]
Does hitting the reset button force the keyboard to register? I had a similar issue when switching inputs using a hub, and hitting reset when switching inputs fixed it.

**Comment by steakscience** [comment]
> Does hitting the reset button force the keyboard to register? I had a similar issue when switching inputs using a hub, and hitting reset when switching inputs fixed it.

Before I try this, RESET will not reset the configuration right?

**Comment by dhanakane** [comment]
No. You need to flash the keymap to do that and that involves putting the keyboard into bootloader mode.

**Comment by steakscience** [comment]
> No. You need to flash the keymap to do that and that involves putting the keyboard into bootloader mode.
> […](#)

I've verified that pressing Reset after losing connection doesn't fix this. I still need to unplug-replug

**Comment by dhanakane** [comment]
Are you using VIA or something else? AFAIK, hitting reset forces the board to re-register.

If you’re using VIA and reset isn’t forcing the board to re-register then I’m out of ideas unfortunately. Maybe check the previous issues reported if there’s a fix?

**Comment by NiciTheNici** [comment]
Did you manage to figure this out?

**Comment by steakscience** [comment]
> Did you manage to figure this out?

Nope
